docs/migrating.md

```diff
@@ -0,0 +1,301 @@
+  how `Yields` worked before.
+- `Args` and `PositionalArgs` have been merged into `Args: { Named: ...; Positional: ... }`. If your component only has
+  named args, the wrapping layer can be skipped and you can continue to use `Args: MyNamedArgs` as before.
+
+{% tabs %}
+{% tab title="Glimmer Component Before" %}
+
+```typescript
+import Component from '@glint/environment-ember-loose/glimmer-component';
+
+export interface MyComponentSignature {
+  Args: { message: string };
+  Yields: { default: [] };
+  Element: HTMLDivElement;
+}
+
+export default class MyComponent extends Component<MyComponentSignature> {
+  // ...
+}
+```
+
+{% endtab %}
+{% tab title="Glimmer Component After" %}
+
+```typescript
+import Component from '@glimmer/component';
+
+export interface MyComponentSignature {
+  Args: { message: string };
+  Blocks: { default: [] };
+  Element: HTMLDivElement;
+}
+
+export default class MyComponent extends Component<MyComponentSignature> {
+  // ...
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+Note that for `EmberComponent` subclasses, there is no native `ArgsFor` equivalent, and the `ArgsFor` helper type will
+be removed in Glint `0.8.0` along with the rest of the `@glint/envrionment-ember-loose/ember-component` module.
+
+You can instead define your named args in a dedicated type declaration, or write a simple `ArgsFor` helper for your own
+project if you wish.
+
+{% tabs %}
+{% tab title="Ember Component Before" %}
+
+```typescript
+import Component, { ArgsFor } from '@glint/environment-ember-loose/ember-component';
+
+export interface MyComponentSignature {
+  Args: { count?: number };
+  PositionalArgs: [message: string];
+}
+
+export default interface MyComponent extends ArgsFor<MyComponentSignature> {}
+export default class MyComponent extends Component<MyComponentSignature> {
+  // ...
+}
+```
+
+{% endtab %}
+{% tab title="Ember Component After" %}
+
+```typescript
+import Component from '@ember/component';
+
+export interface MyComponentNamedArgs {
+  count?: number;
+}
+
+export interface MyComponentSignature {
+  Args: {
+    Named: MyComponentNamedArgs;
+    Positional: [message: string];
+  };
+}
+
+export default interface MyComponent extends MyComponentNamedArgs {}
+export default class MyComponent extends Component<MyComponentSignature> {
+  // ...
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+#### Helpers
+
+For helpers, `NamedArgs` and `PositionalArgs` have been merged into `Args: { Named: ...; Positional: ... }`, similar to
+the change for components. Unlike components, however, since neither type of argument is privileged over the other in
+the way helpers are defined today, there is no shorthand.
+
+For helpers and modifiers, NamedArgs and PositionalArgs have been merged into Args: { Named: ...; Positional: ... } similar to the change for components. However, since neither named nor positional args are more privileged in today's implementation of helpers and modifiers, there is no shorthand.
+
```

Comment:
```suggestion
```

Reply:
Oops! I had this notion in the back of my brain that I'd never deleted that after I was done copying from it, but then it wasn't where I looked 😂